### PR TITLE
Fix custom roles functionality

### LINKED ISSUE #2754: Custom roles functionality broken
As seen below, I created a custom role superman with access to modules,reports and messages panels.
selection_035

I assigned the custom role to a different user.
selection_034

But the user is still not able to access the specified panels.
selection_036

Steps to recreate the issue:
- Create a custom role
- Assign that role to a different user
- After assigning, try to access the respective assigned panels from the different user account

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #2754 .

selection_037

### CODE REVIEW

File: app/models/user.py [hunk 1]
```diff
@@ -220,8 +220,8 @@ def can_access_panel(self, panel_name):
             return True
 
         custom_sys_roles = UserSystemRole.query.filter_by(user=self)
-        for role in custom_sys_roles:
-            if role.can_access(panel_name):
+        for crole in custom_sys_roles:
```
Comment (SaptakS) [inline app/models/user.py]: Use a better name such as `custom_role`

File: app/views/super_admin/__init__.py [hunk 2]
```diff
@@ -35,5 +35,5 @@ def check_accessible(panel_name):
     if not AuthManager.is_accessible():
         return redirect(url_for('admin.login_view', next=request.url))
     else:
-        if not current_user.can_access_panel(panel_name) or not current_user.is_staff:
+        if not current_user.can_access_panel(panel_name) and not current_user.is_staff:
```
Comment (SaptakS) [inline app/views/super_admin/__init__.py]: Why should this be `and` instead of `or`?? Can you elaborate?

Comment (shubham-padia) [inline app/views/super_admin/__init__.py]: `current_user.is_staff` determines whether the user is an admin(with access to all panels) or not and `current_user.can_access_panel(panel_name)` checks if the user has access to that specific panel or not. Thus if a sales admin wants to access only the sales tab he will not be able to do so as `current_user.is_staff` will be false for him/her.

File: app/api/helpers/custom_fields.py [hunk 3]
```diff
@@ -26,6 +26,7 @@ def format(self, value):
         format the text in database for output
         works only for GET requests
         """
+        print(value)
```
Comment (SaptakS) [inline app/api/helpers/custom_fields.py]: Remove this print statement.

File: app/api/helpers/special_fields.py [hunk 4]
```diff
@@ -27,17 +27,9 @@ class EventSubTopicField(fields.ChoiceString):
 
     def __init__(self, **kwargs):
         super(EventSubTopicField, self).__init__(
-            choice_list=[],
+            choice_list=DataGetter.get_event_subtopics().get(topic, []),
             **kwargs)
 
-    def validate(self, value):
```
Comment (SaptakS) [inline app/api/helpers/special_fields.py]: Can you explain removing this?

Comment (shubham-padia) [inline app/api/helpers/special_fields.py]: @Saptaks Sorry, I was testing some other things and accidentally pushed them. Same for the print statement

**Comment by shubham-padia** [comment]
@SaptakS Can you please review again :smile: . Thanks.

**Comment by SaptakS** [comment]
@niranjan94 can you please look into this PR. Thanks.

**Pull request merged**